Rd Framework api changes (#1271)
JetBrains/intellij-sdk-docs: reference_guide/api_changes_list_2024.md

### JavaScript Plugin 2024.1
`com.intellij.lang.javascript.documentation.JSDocumentationProvider.generateDoc(PsiElement, PsiElement)` method marked final
: Override `com.intellij.lang.javascript.documentation.JSDocumentationProvider.generateDoc(PsiElement, PsiElement, Ref<String>)` instead

### Rd Framework 2024.1

`com.jetbrains.rd.framework.IMarshaller.DefaultImpls.getId(IMarshaller)` method removed

> : Due to `RdId` becoming a value class, the `getId` method is removed at runtime, causing unresolved method invocations. Use the method that returns `long` in java and just recompile the kotlin code

Reviewer: ```suggestion
: Due to RdId becoming a value class, the getId method is removed at runtime, causing unresolved method invocations. Use the method that returns long in Java and recompile the Kotlin code.
```

`com.jetbrains.rd.framework.RdId.write(AbstractBuffer)` method removed

> : Due to `RdId` becoming a value class, the `getId` method is removed at runtime, causing unresolved method invocations. Use `AbstractBuffe::writeLong(long)` method in java and just recompile the kotlin code.

Reviewer: ```suggestion
: Due to `RdId` becoming a value class, the `getId` method is removed at runtime, causing unresolved method invocations. Use `AbstractBuffer.writeLong(long)` method in Java and recompile the Kotlin code.
```

`org.digma.intellij.plugin.rider.protocol.LensPerObjectId.getRdid()` method removed

> : Due to `RdId` becoming a value class, the `getId` method is removed at runtime, causing unresolved method invocations. Use the method that returns `long` in java and just recompile the kotlin code

Reviewer: ```suggestion
: Due to `RdId` becoming a value class, the `getId` method is removed at runtime, causing unresolved method invocations. Use the method that returns `long` in Java and recompile the Kotlin code.
```

`com.jetbrains.rd.framework.RdId.Companion.read(AbstractBuffer)` method removed

> : Due to `RdId` becoming a value class, the `getId` method is removed at runtime, causing unresolved method invocations. Use `AbstractBuffe::readLong()` method in java and just recompile the kotlin code.

Reviewer: ```suggestion
: Due to `RdId` becoming a value class, the `getId` method is removed at runtime, causing unresolved method invocations. Use `AbstractBuffer.readLong()` method in Java and recompile the Kotlin code.
```

`com.jetbrains.rd.framework.base.RdBindableBaseKt.withId(RdBindableBase, RdId)` method removed

> : Due to `RdId` becoming a value class, the `getId` method is removed at runtime, causing unresolved method invocations. Use `withId(RdBindableBase, long)` method in java and just recompile the kotlin code.

Reviewer: ```suggestion
: Due to `RdId` becoming a value class, the `getId` method is removed at runtime, causing unresolved method invocations. Use `withId(RdBindableBase, long)` method in Java and recompile the Kotlin code.
```

> `com.jetbrains.rd.framework.IMarshaller.getId-yyTGXKE()` abstract method added

Reviewer: Plugin Verifier is not able to parse such a method name (and we won't be able to fix it before the release). Please remove this entry for now.
